### ISSUE TITLE
Always show first and last ticks?

### ISSUE THREAD

**Issue by frnsys** [feral-dot-io/leptos-chartistry#30]
Hey, thanks for this library. Is there an option to always show the first and last ticks?

**Comment by feral-dot-io** [comment]
Does Series [with_min_x](https://docs.rs/leptos-chartistry/latest/leptos_chartistry/struct.Series.html#method.with_min_x) and with_max_x (and also for _y) work for you?

**Comment by frnsys** [comment]
Unfortunately no, manually setting hasn't changed it:

![image](https://github.com/user-attachments/assets/c07fcb92-cb6c-4717-b3a5-d2a9b3dffac2)

**Comment by feral-dot-io** [comment]
Thanks for the image. To be clear, is it the date ticks that you want shown first and last? What values do you want to them to show?

If it's 2010 and 2023 then they will spill over the edges and could overlap labels on the side.

If it's 2011 and 2022 then there might not be enough space for the text plus padding. Can you share an image with `<Chart debug=true ... />` set? It will give an idea of the spacing involved.

**Comment by frnsys** [comment]
Yes, sorry for not clarifying, it's the X-axis/date ticks. The actual min and maxes are 2010 and 2023. Is there some way to set it so that the first and last ticks are left and right anchored, respectively?

Here it is with debug enabled:

![image](https://github.com/user-attachments/assets/b349022e-c553-4883-943d-8d3addb4ce6e)

**Comment by feral-dot-io** [comment]
Can you share the full <Chart> call? There's some extra spacing inside the tick labels that probably shouldn't be there.

The other thing is that the date picking algorithm is trying to pick the most relevant periods (in this case years) and then it's trying to fit as many as possible. I think this is done by having a preference for the end ranges of the data although my memory is a little fuzzy on this.

The issue with what you're asking for is that the second half of the "2023" label would leak out into right edge and then overlap the kWh label. The right tick labels are kind of doing it already so I'm thinking it might be worthwhile making this choice configurable (extend labels into the corner space) and making it the default for x ticks which are most likely to be dates.
